asv_bench/benchmarks/strftime.py

```diff
@@ -45,6 +48,44 @@ def time_frame_datetime_formatting_custom(self, obs):
         self.data["dt"].dt.strftime(date_format="%Y-%m-%d --- %H:%M:%S")
 
 
+class PeriodStrftime:
+    timeout = 1500
+    params = ([1000, 10000], ["D", "H"])
+    param_names = ["obs", "fq"]
+
+    def setup(self, obs, fq):
+        self.data = pd.DataFrame(
+            {
+                "p": pd.period_range(start="2000-01-01", periods=obs, freq=fq),
+                "r": [np.random.uniform()] * obs,
+            }
+        )
+
+    def time_frame_period_to_str(self, obs, fq):
+        self.data["p"].astype(str)
+
+    def time_frame_period_formatting_default(self, obs, fq):
+        """Note that as opposed to datetimes, the default format of periods are
+        many and depend from the period characteristics, so we have almost no chance
+        to reach the same level of performance if a 'default' format string is
+        explicitly provided by the user. See
+        time_frame_datetime_formatting_default_explicit above."""
+        self.data["p"].dt.strftime(date_format=None)
+
+    def time_frame_period_formatting_index_default(self, obs, fq):
+        self.data.set_index("p").index.format()
```

Comment:
this is going to conflate the timing of format with set_index.  is there a viable alternative?

Reply:
ok fixed